sandbox/libs/authn/src/main/java/org/opensearch/authn/Permission.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.authn;
+
+import java.io.IOException;
+import java.util.Objects;
+
+public class Permission {
+    private final static String PERMISSION_DELIMITER = "\\.";  
+    private final String[] permissionChunks; 
+
+    public Permission(final String permission) {
+        try {
+            this.permissionChunks = permission.split(PERMISSION_DELIMITER);
+        } catch (Exception) {
+            throw new InvalidPermissionName(permission);
+        }
+    }
+
+    public boolean matches(final String permissionRequired) {
+        Objects.nonNull(permissionRequired);
+
+        final Permission required = new Permission(permissionRequired);
+        for(int i = 0; i < this.permissionChunks.length; i++) {
+            if (!this.permissionChunks[i].equals(required.permissionChunks[i])) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    public static void checkIsValid(final String permission) {
+        new Permission(permission);
```

Comment:
Instead of creating a new object to check permission validity here can we introduce following:

```java
public Permission(final String permission) {
       this.permissionChunks = splitIntoChunks(permission);
}

```

```java
private static String[] splitIntoChunks(final String permission){
        try {
            return permission.split(PERMISSION_DELIMITER);
        } catch (Exception) {
            throw new InvalidPermissionName(permission);
        }
}
```
```java
public static void checkIsValid(final String permission) {
        splitIntoChunks(permission);
}
```